src/app/shared/functions/user.ts

```diff
@@ -0,0 +1,42 @@
+import prisma from '@db/prismaSingleton'
+
+import type { KindeUser } from '@kinde-oss/kinde-auth-nextjs/dist/types'
+
+import type { User } from '@prisma/client'
+
+import { createUser } from '@actions/user'
+
+export const findOrCreateUser = async (externalUser: KindeUser | null) => {
+  if (!externalUser) return null
+
+  let user: User | null
+
+  user = await prisma.user.findFirst({
+    where: {
+      userAccounts: {
+        some: {
+          provider: 'Kinde',
```

Comment:
currently this is hardcoded with Kinde, if we were ever to switch to some other auth provider we would need to  refactor here

Reply:
will do for now